src/dswx_sar/dswx_comparison.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+import os
+import argparse
+import numpy as np
+from osgeo import gdal
+
+COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two DSWx-SAR products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input images')
+
+    return parser
+
+def _get_prefix_str(flag_same, flag_all_ok):
+    flag_all_ok[0] = flag_all_ok[0] and flag_same
+    return '[OK]   ' if flag_same else '[FAIL] '
+
+def _print_first_value_diff(image_1, image_2, prefix):
+    """
+    Print first value difference between two images.
+
+       Parameters
+       ----------
+       image_1 : numpy.ndarray
+            First input image
+       image_2: numpy.ndarray
+            Second input image
+       prefix: str
+            Prefix to the message printed to the user
+    """
+    flag_error_found = False
+    for i in range(image_1.shape[0]):
+        for j in range(image_1.shape[1]):
+            if (abs(image_1[i, j] - image_2[i, j]) <=
+                    COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE):
+                continue
+            print(prefix + f'     * input 1 has value'
+                  f' "{image_1[i, j]}" in position'
+                  f' (x: {j}, y: {i})'
+                  f' whereas input 2 has value "{image_2[i, j]}"'
+                  ' in the same position.')
+            flag_error_found = True
+            break
+        if flag_error_found:
+            break
```

Comment:
```suggestion
            if (abs(image_1[i, j] - image_2[i, j]) >=
                    COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE):

            print(prefix + f'     * input 1 has value'
                  f' "{image_1[i, j]}" in position'
                  f' (x: {j}, y: {i})'
                  f' whereas input 2 has value "{image_2[i, j]}"'
                  ' in the same position.')
            break
```

Reply:
Maybe you should add `flag_error_found = True` right after the if statement to avoid that. But I'm fine with your way of coding. The code is not complex, and okay to follow.